### ISSUE TITLE
`gosec` `G601: Implicit memory aliasing in for loop` linting errors

### ISSUE THREAD

**Issue by atc0005** [atc0005/check-vmware#916]
## Overview

From the latest CI linting results:

```log
internal/vsphere/snapshots.go:893:46: G601: Implicit memory aliasing in for loop. (gosec)
				listFunc(vm, snapTree.ChildSnapshotList, &snapTree.Snapshot)
				                                         ^
internal/vsphere/snapshots.go:1154:47: G601: Implicit memory aliasing in for loop. (gosec)
				crawlFunc(vm, snapTree.ChildSnapshotList, &snapTree.Snapshot)
```

From using the experimental 1.21 GC flag:

```log
# github.com/atc0005/check-vmware/internal/vsphere
internal/vsphere/snapshots.go:877:10: loop variable snapTree now per-iteration, heap-allocated
internal/vsphere/snapshots.go:984:10: loop variable snapTree now per-iteration, heap-allocated
# github.com/atc0005/check-vmware/cmd/check_vmware_hs2ds2vms
cmd/check_vmware_hs2ds2vms/main.go:297:14: loop variable pairing now per-iteration, stack-allocated
```

Note the last item was not caught by the `gosec` linter.

This output obtained via:

```shell
podman container run --platform linux/amd64 --rm -it -v $PWD:/code -w /code golang:1.21.0
go build -gcflags=all=-d=loopvar=2 ./...
```

## References

- https://stackoverflow.com/questions/62446118/implicit-memory-aliasing-in-for-loop
- https://github.com/golang/go/wiki/LoopvarExperiment#can-this-change-break-programs
- https://github.com/atc0005/shared-project-resources/issues/145
